src/network/service.rs

```diff
@@ -5,10 +5,12 @@ use crate::intercom::{
     ReplyStream, SubscriptionFuture, SubscriptionStream, TransactionMsg,
 };
 use crate::utils::task::TaskMessageBox;
+use std::collections::BTreeMap;
 use std::marker::PhantomData;
 
+use network::p2p_topology::{Gossip, Id, P2pTopology};
 use network_core::{
-    gossip::NodeId,
+    gossip::{self, Gossip as G},
```

Comment:
Could better do `Gossip as GossipTrait`.
Or do `<Gossip as gossip::Gossip>` where trait disambiguation is needed.
Or name our concrete gossip type `PoldercastGossip` to avoid ambiguity.